src/azure-cli/azure/cli/command_modules/identity/_client_factory.py

```diff
@@ -4,10 +4,14 @@
 # --------------------------------------------------------------------------------------------
 
 
-def _msi_client_factory(cli_ctx, **_):
+def _msi_client_factory(cli_ctx, api_version=None, **_):
     from azure.cli.core.profiles import ResourceType
     from azure.cli.core.commands.client_factory import get_mgmt_service_client
-    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_MSI)
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_MSI, api_version=api_version)
+
+
+def _msi_list_resources_client(cli_ctx, **_):
+    return _msi_client_factory(cli_ctx, api_version='2021-09-30-preview').user_assigned_identities
```

Comment:
Please add a comment to explain why we need to specify the ad-hoc api-version

Reply:
Ad-hoc API version can leads to `scripts/trim_sdk.py` script failure.

It should be declared at

https://github.com/Azure/azure-cli/blob/085daccc19445387586e4f532b325f51e526eaff/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L411-L428

Reply:
Hi @jiasli, declaration has been added here. Could you please review this change?